src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/value_control_form.test.tsx

```diff
@@ -264,6 +267,34 @@ describe('ValueControlForm', () => {
         // values preview panel should be rendered
         expect(await findByTestId('esqlValuesPreview')).toBeInTheDocument();
       });
+
+      it('should be able to change in fields type', async () => {
+        const { findByTestId } = render(
+          <IntlProvider locale="en">
+            <ESQLControlsFlyout
+              initialVariableType={ESQLVariableType.VALUES}
+              queryString="FROM foo | WHERE field =="
+              onSaveControl={jest.fn()}
+              closeFlyout={jest.fn()}
+              onCancelControl={jest.fn()}
+              search={searchMock}
+              esqlVariables={[]}
+            />
+          </IntlProvider>
+        );
+        // variable name input should be rendered and with the default value
+        expect(await findByTestId('esqlVariableName')).toHaveValue('?field');
+        // change the variable name to ?value
+        const variableNameInput = await findByTestId('esqlVariableName');
+        fireEvent.change(variableNameInput, { target: { value: '??field' } });
```

Comment:
Generally it's best to prefer the `userEvent` library. It does a better job of simulating actual user behavior.

Reply:
Noted, thanx for the tip 🙌